src/Beman/Example/tests/example.t.cpp

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <Beman/Example/example.hpp>
+
+#include <gtest/gtest.h>
+
+TEST(ExampleTest, call_identity) {
+    for (int i = -100; i < 100; ++i) {
+        EXPECT_EQ(i, beman::example::identity()(i));
+    }
+}
+
+// Test comparing beman::example::identity with std::identity.
+// Requires std::identity support.
+#if defined(__cpp_lib_identity)
+#include <functional>
```

Comment:
I suggest putting all includes at the top and making them unconditional. That simplifies this code.